### PR TITLE
Fix issues with undo and cloning

### PR DESCRIPTION
**Pull request by rockfordlhotka**
#1212 Fix issues with undo and cloning

### CODE REVIEW

File: Source/Csla.Shared/Core/FieldManager/FieldDataManager.cs [hunk 1]
```diff
@@ -1,7 +1,7 @@
 ﻿//-----------------------------------------------------------------------
 // <copyright file="FieldDataManager.cs" company="Marimer LLC">
 //     Copyright (c) Marimer LLC. All rights reserved.
-//     Website: https://cslanet.com
+//     Website: http://cslanet.com
```
Comment (Hadronicus) [inline Source/Csla.Shared/Core/FieldManager/FieldDataManager.cs]: The website link was reverted to http. This was probably a mistake.

File: Source/Csla.Shared/Core/FieldManager/FieldDataManager.cs [hunk 2]
```diff
@@ -756,21 +688,15 @@ protected override void OnSetState(SerializationInfo info, StateMode mode)
...
         {
-          //string xml = info.GetValue<string>("_stateStack");
-          byte[] xml = info.GetValue<byte[]>("_stateStack");
-          MobileList<SerializationInfo> list = (MobileList<SerializationInfo>)MobileFormatter.Deserialize(xml);
-          SerializationInfo[] layers = list.ToArray();
-          Array.Reverse(layers);
-          foreach (SerializationInfo layer in layers)
-            _stateStack.Push(layer);
+          var stackArray = info.GetValue<byte[][]>("_stateStack");
+          _stateStack.Clear();
```
Comment (Hadronicus) [inline Source/Csla.Shared/Core/FieldManager/FieldDataManager.cs]: _stateStack.Clear(); is being called twice in the same control flow. Once just before the **if** statement and once inside it. It looks like this is probably just a copy/past mistake from the implementation in UndoableBase.

I'm not sure if the call before the **if** statement or the one inside is more appropriate.

File: Source/Csla.Shared/Core/FieldManager/FieldDataManager.cs (hunk 1 above)
Comment (rockfordlhotka) [reply Source/Csla.Shared/Core/FieldManager/FieldDataManager.cs]: Thanks, will be fixed in another PR

File: Source/Csla.Shared/Core/FieldManager/FieldDataManager.cs (hunk 2 above)
Comment (rockfordlhotka) [reply Source/Csla.Shared/Core/FieldManager/FieldDataManager.cs]: Thanks, will be fixed in another PR

**Pull request merged**